ddl/backfilling.go

```diff
@@ -946,6 +1041,316 @@ func injectCheckBackfillWorkerNum(curWorkerSize int, isMergeWorker bool) error {
+		return nil
+	}
+
+	if err := dc.isReorgRunnable(reorgInfo.Job.ID); err != nil {
+		return errors.Trace(err)
+	}
+
+	currBackfillJobID := int64(1)
+	err := checkAndHandleInterruptedBackfillJobs(sess, reorgInfo.Job.ID, reorgInfo.currElement.ID, reorgInfo.currElement.TypeKey)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	maxBfJob, err := GetMaxBackfillJob(sess, reorgInfo.Job.ID, reorgInfo.currElement.ID, reorgInfo.currElement.TypeKey)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if maxBfJob != nil {
+		startKey = maxBfJob.EndKey
+		currBackfillJobID = maxBfJob.ID + 1
+	}
+
+	var isUnique bool
+	if bfWorkerType == typeAddIndexWorker {
+		idxInfo := model.FindIndexInfoByID(t.Meta().Indices, reorgInfo.currElement.ID)
+		isUnique = idxInfo.Unique
+	}
+	err = dc.splitTableToBackfillJobs(sess, reorgInfo, t, isUnique, bfWorkerType, startKey, currBackfillJobID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	var backfillJobFinished bool
+	jobID := reorgInfo.Job.ID
+	ticker := time.NewTicker(300 * time.Millisecond)
+	defer ticker.Stop()
+	for {
+		if err := dc.isReorgRunnable(reorgInfo.Job.ID); err != nil {
+			return errors.Trace(err)
+		}
+
+		select {
+		case <-ticker.C:
+			if !backfillJobFinished {
+				err := checkAndHandleInterruptedBackfillJobs(sess, jobID, reorgInfo.currElement.ID, reorgInfo.currElement.TypeKey)
+				if err != nil {
+					logutil.BgLogger().Warn("[ddl] finish interrupted backfill jobs", zap.Int64("job ID", jobID), zap.Error(err))
+					return errors.Trace(err)
+				}
+
+				bfJob, err := getBackfillJobWithRetry(sess, BackfillTable, jobID, reorgInfo.currElement.ID, reorgInfo.currElement.TypeKey, false)
+				if err != nil {
+					logutil.BgLogger().Info("[ddl] getBackfillJobWithRetry failed", zap.Int64("job ID", jobID), zap.Error(err))
+					return errors.Trace(err)
+				}
+				if bfJob == nil {
+					backfillJobFinished = true
+					logutil.BgLogger().Info("[ddl] finish backfill jobs", zap.Int64("job ID", jobID))
+				}
+			}
+			if backfillJobFinished {
+				// TODO: Consider whether these backfill jobs are always out of sync.
+				isSynced, err := checkJobIsSynced(sess, jobID)
+				if err != nil {
+					logutil.BgLogger().Warn("[ddl] checkJobIsSynced failed", zap.Int64("job ID", jobID), zap.Error(err))
+					return errors.Trace(err)
+				}
+				if isSynced {
+					logutil.BgLogger().Info("[ddl] sync backfill jobs", zap.Int64("job ID", jobID))
+					return nil
+				}
+			}
+		case <-dc.ctx.Done():
+			return dc.ctx.Err()
+		}
+	}
+}
+
+func checkJobIsSynced(sess *session, jobID int64) (bool, error) {
+	var err error
+	var unsyncedInstanceIDs []string
+	for i := 0; i < retrySQLTimes; i++ {
+		unsyncedInstanceIDs, err = getUnsyncedInstanceIDs(sess, jobID, "check_backfill_history_job_sync")
+		if err != nil {
+			return false, errors.Trace(err)
+		}
+		if len(unsyncedInstanceIDs) == 0 {
+			return true, nil
+		}
+
+		logutil.BgLogger().Info("[ddl] checkJobIsSynced failed", zap.Int("tryTimes", i), zap.Error(err))
```

Comment:
```suggestion
		logutil.BgLogger().Info("[ddl] checkJobIsSynced failed", zap.Int("tryTimes", i))
```

The error is always `nil`.